### ISSUE TITLE
Whenever I set "multi_label = True", the training gets stuck at 1 epoch

### ISSUE THREAD

**Issue by ParthJawale1996** [flairNLP/flair#838]
A clear and concise description of what you want to know.
I'm doing a multiclass text classification. Whenever I set the "multi_label" argument as True, the training gets stuck at the first epoch. If I remove the multi_label = True (or well, make it False), it seems to run seamlessly. I thought this could be a memory issue, so I tried to use just the "glove" embeddings instead the heavy-duty I was trying before, but it still is getting stuck at the first epoch. 
What do I do? Any ideas are welcome at this point. 

Yes, I have the latest release 0.4.2, and trying to work on Jupyter notebooks.

**Comment by ParthJawale1996** [comment]
@alanakbik Do you have any insights on this problem? I've been trying quite a few things but to no avail.

**Comment by alanakbik** [comment]
@ParthJawale1996 difficult to say - I run ran a few experiments with `multi_label` set to `True` or `False` but I am not encountering this issue. Could it be dataset related? How are you reading in the data?

**Comment by ParthJawale1996** [comment]
@alanakbik I'm pasting my script here:

import os
from flair.data_fetcher import NLPTaskDataFetcher
from flair.embeddings import FlairEmbeddings, DocumentPoolEmbeddings, WordEmbeddings, DocumentRNNEmbeddings, StackedEmbeddings
from flair.data import Sentence
from flair.training_utils import EvaluationMetric
from flair.models import TextClassifier
from flair.datasets import ClassificationCorpus
from flair.trainers import ModelTrainer
from pathlib import Path
import gensim

stacked_embeddings = [WordEmbeddings('glove')]
document_embeddings = DocumentRNNEmbeddings(stacked_embeddings, hidden_size=512, reproject_words=True, reproject_words_dimension=256)
corpus = NLPTaskDataFetcher.load_classification_corpus(Path('./'), test_file='CombinedTestSet.txt', train_file='CombinedTrainingSet.txt')

classifier = TextClassifier(document_embeddings, label_dictionary=corpus.make_label_dictionary(), multi_label = True)

trainer = ModelTrainer(classifier, corpus)

trainer.train('./',
                   learning_rate=0.1,
                   mini_batch_size=32,
                   anneal_factor=0.5,
                   patience=5,
                   max_epochs=10,
                   embeddings_in_memory=False)

![image](https://user-images.githubusercontent.com/46942720/60194196-b6b6fe00-97ed-11e9-8a09-7e18c612c611.png)
This is sort of what my dataframe for the training looks like.

**Comment by ParthJawale1996** [comment]
multi_label = "True" would be necessary because my data has a lot of labels, so setting it to "false" would not work.

**Comment by alanakbik** [comment]
Could you print out: 

```python
label_dict = corpus.make_label_dictionary()

print(label_dict.multi_label)
```

If the value is `False`, it means that there is no data point that has more than one label in your dataset. You only should set multi_label to True if a single data point can have multiple labels.

**Comment by ParthJawale1996** [comment]
It prints out a list of labels. 
[b'GettyImages-101884760_super.jpg', b'GettyImages-102071017_super.jpg', b'GettyImages-102416416_super.jpg', b'GettyImages-103060530_super.jpg', b'GettyImages-103332256_super.jpg', b'GettyImages-103576254_super.jpg', b'GettyImages-103585631_super.jpg', b'GettyImages-105759955_medium.jpg'.....]

Why is it appending a 'b to all the label names?
Also, it prints out **False**.

**Comment by ParthJawale1996** [comment]
So I will train using multi_label = False, but from yesterday's experiments I remember that when I created a model and tried to predict using best_model.pt and gave in a string as an input, I got "str object has no attribute "token"". Any idea why that might be?

**Comment by alanakbik** [comment]
You need to always make a `Sentence` object for the text you want to predict. See the [tutorial](https://github.com/zalandoresearch/flair/blob/master/resources/docs/TUTORIAL_2_TAGGING.md#tagging-with-pre-trained-sequence-tagging-models).

**Comment by ParthJawale1996** [comment]
If I've not explicitly made a dev file, and only passed "train" and "test", what will it do? I think it would still make some sort of a dev file, if I'm not wrong?

**Comment by alanakbik** [comment]
Yes it will automatically sample a dev file from the training data.

**Comment by ParthJawale1996** [comment]
So, I actually have my custom evaluation method which requires me to predict on the entire text column in my dataset. So I can pass all these text chunks as a list, right? As shown in a way similar to [here:tagging a list of sentences](https://github.com/zalandoresearch/flair/blob/master/resources/docs/TUTORIAL_2_TAGGING.md#tagging-a-list-of-sentences)

**Comment by Barath19** [comment]
hey @alanakbik, what is the seperator if its a multilabel eg : , or /
